api/api-cache.php

```diff
@@ -0,0 +1,132 @@
+            return $r;
+        }
+        return false;
+    }
+    return null;
+}
+
+/**
+ * Adds or updates an entry into the cache table.
+ * @param string $endpoint The endpoint for the API request
+ * @param string $content The content of API request's response
+ * @param int|string $config The ID of the config or its name in the config table
+ */
+function insertCached(string $endpoint, string $content, $config)
+{
+    global $db;
+    $r = getFromApiCacheTable($endpoint);
+    $queryString = "";
+    if ($r) {
+        $id = $r['id'];
+        $queryString = 'UPDATE api_cache 
+                        SET last_successfully_retrieved = UTC_TIMESTAMP, 
+                            last_attempted_retrieval = UTC_TIMESTAMP, 
+                            retry_count = 0, 
+                            content =\'' . $db->real_escape_string($content) . '\'
+                        WHERE id = ' . $id;
+
+    } else {
+        $configId = 1;
+        if (gettype($config) == 'integer') {
+            $configId = $config;
+        } else if (gettype($config) == 'string') {
+            $configId = $db->query('SELECT * FROM outgoing_request_cache_config WHERE config_name = "' . $db->real_escape_string($config) . '"')->fetch_array(MYSQLI_ASSOC)['id'];
+        }
+        $queryString = "INSERT INTO api_cache (endpoint, last_successfully_retrieved, last_attempted_retrieval, config_id, content) 
+                    values ('" . $db->real_escape_string($endpoint) . "', UTC_TIMESTAMP,
+                            UTC_TIMESTAMP," . $configId . ", '" . $db->real_escape_string($content) . "')";
+    }
+    $q = $db->query($queryString);
+    if ($q === false) {
+        error_log('Failed to update API cache: ' . $db->error);
+    }
+}
+
+/**
+ * Deletes an entry from the cache or deletes all entries from the cache.
+ * @param string|null $endpoint The endpoint of the API request to be removed. If null, empties the cache
+ */
+function clearCached(string $endpoint = null){
+    global $db;
+    if ($endpoint) {
+        $queryString = "DELETE FROM api_cache WHERE endpoint = '" . $db->real_escape_string($endpoint) . "'";
+    } else{
+        $queryString = "DELETE FROM api_cache";
+    }
+    $q = $db->query($queryString);
+    if ($q === false) {
+        error_log("Failed to clear API cache: " . $db->error);
+    }
+}
+
+/**
+ * Adds a new cache config to the config table.
+ * @param int $ttl The time to live for cached entries, measured in seconds
+ * @param string|null $config_name (optional) The name of the config
+ * @return bool|mysqli_result The results of the MySQLi query
+ */
+function addNewCacheConfig(int $ttl, $config_name = null)
+{
+    global $db;
+    if ($config_name) {
+        return $db->query('INSERT INTO outgoing_request_cache_config (ttl, config_name) VALUES (' . $ttl . ', "' . $db->real_escape_string($config_name) . '")');
+    }
+    return $db->query('INSERT INTO outgoing_request_cache_config (ttl) VALUES (' . $ttl . ')');
```

Comment:
`config_name` is always required, we should not allow it to be inserted empty, in fact, we should update the SQL to make it a non nullable column